src/backend/functions/orchestrator/lib/index.js

```diff
@@ -2,28 +2,40 @@ const AWS = require('aws-sdk');
 const R = require('ramda');
 const converter = AWS.DynamoDB.Converter;
 
+// o is similar to R.compose but takes exactly two arguments
+const {o} = R;
+
+// compose that works on promise returning functions
+const composeP = (...fns) => R.composeWith(R.then)(fns);
+
 // snakeToCamel :: String -> String
-const snakeToCamel = R.replace(/([_][a-z])/g, R.compose(R.replace('_', ''), R.toUpper));
+const snakeToCamel = R.replace(/([_][a-z])/g, o(R.replace('_', ''), R.toUpper));
 
 function noop() {}
 
 // convertEnvVars :: {k: v} -> {k: v}
 const convertEnvVars = R.pipe(
     R.toPairs,
-    R.map(R.adjust(0, R.compose(snakeToCamel, R.toLower))),
+    R.map(R.adjust(0, o(snakeToCamel, R.toLower))),
     R.fromPairs,
     R.evolve({subnets: subnets => [...subnets.split(',').map(R.trim)]})
 );
 
+const stopTranscription = R.curry((ecs, {mediaUrl, cluster, task, tasksTableName}) => {
+   return ecs.stopTask({cluster, task})
```

Comment:
is there a possible (albeit very unlikely) race condition here of the ECS task stopping before the TERMINATING state is written to DDB e.g. the function faces some latency writing to DDB? e.g. stoptask issues SIGTERM -> shutdown hook writes TERMINATED status -> function writes TERMINATING

Reply:
feels like either the orchestrator or the lpc should be managing job state rather than both otherwise this sort of situation could arise?

Reply:
But how else can the container singal that it's shutdown? The orchestrator lambda is only triggered on a write to the DB so there's no way for it to manage that part of the lifecycle.

Regarding the race conditon, the TERMINATING condition is triggered by the user clicking on the stop button on the frontend, this triggers the lambda to stop the task so the write to the DB precedes any interaction with ECS.

Reply:
yeah you're right, got the flow wrong in my head - makes sense now